common/src/disk.rs

```diff
@@ -583,3 +647,24 @@ impl TryFrom<i64> for M2Slot {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_strategy::proptest;
+
+    #[proptest]
+    fn parse_dataset_name(pool_id: [u8; 16], kind: DatasetKind) {
+        let pool_id = ZpoolUuid::from_bytes(pool_id);
```

Comment:
arguably this could be producing invalid UUIDv4s - e.g., the version/variant bits could be wrong, with an arbitrary byte input sequence - but I dunno if we care.